python/lsst/meas/base/afterburner.py

```diff
@@ -0,0 +1,183 @@
+    registry = PluginRegistry(AfterburnerPluginConfig)
+    ConfigClass = AfterburnerPluginConfig
+
+    def __init__(self,config, name, schema, metadata):
+        """!
+        Initialize the afterburner plugin
+
+        @param[in] config     An instance of afterburner config class.
+        @param[in] name       The string the plugin was registered with.
+        @param[in,out] schema The source schema, New fields should be added here to
+                              hold output produced by this plugin.
+        @param[in] metadata   Plugin metadata that will be attached to the output catalog
+        """
+        BasePlugin.__init__(self, config, name)
+
+    @classmethod
+    def getExecutionOrder(cls):
+        ''' Sets the relative order of plugins (smaller numbers run first).
+
+        Afterburner plugins must run with BasePlugin.DEFAULT_AFTERBURNER or higher
+
+        All plugins must implement this method with an appropriate run level
+        '''
+        raise NotImplementedError()
+
+    @property
+    def PlugType(self):
+        '''
+        This defines if the plugin operates on a single source at a time, or expects the whole catalog.
+        The value defaults to single for a single source, set to multi when the plugin expects the whole
+        catalog. If The plugin is of type multi, the fail method should be implemented to accept the whole
+        catalog. If the plugin is of type the fail method should accept a single source record.
+        '''
+        return 'single'
+
+    def burn(self, cat, **kwargs):
+        """!
+        Process either a single catalog enter or the whole catalog and produce output defined by the plugin
+
+        @param[in,out] cat  Either a lsst source catalog or a catalog entery depending on the plug type
+                            specified in the classes configuration. Results may be added to new columns,
+                            or existing entries altered.
+        @param[in] kwargs   Any additional kwargs that may be passed through the afterburner task.
+        """
+        raise NotImplementedError()
+
+class AbContext(object):
+    '''
+    Context manager to handle catching errors that may have been thrown in an afterburner plugin
+    @param[in] plugin   The plugin that is to be run
+    @param[in] cat      Either a catalog or a source record entry of a catalog, depending of the plugin type,
+                        i.e. either working on a whole catalog, or a single record. 
+    @param[in] log      The log which to write to, most likely will always be the log (self.log) of the object
+                        in which the context manager is used.
+    '''
+    def __init__(self, plugin, cat, log):
+        self.plugin = plugin
+        self.cat = cat
+        self.log = log
+
+    def __enter__(self):
+        return
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        if exc_type is None:
+            return True
+        if exc_type in FATAL_EXCEPTIONS:
```

Comment:
`if issubclass(exc_type, FATAL_EXCEPTIONS):`
and (below)
`elif issubclass(exc_type, MeasurementError):`

As I've stated elsewhere, I think you want to treat anything that inherits from `BaseException` but not from `Exception` as one of the fatal exceptions because those are meant to bring the system down.